src/Compilers/CSharp/Test/Syntax/LexicalAndXml/PreprocessorTests.cs

```diff
@@ -2962,6 +2993,71 @@ private void CheckDiagnosticStringFileName(string compilationFileName, string li
             Assert.Equal(expectedErrorStringFileName, actualErrorStringFileName);
         }
 
+        [Fact]
+        public void TestErrorWithVersion()
+        {
+            var text = "#error version";
+            var node = Parse(text, SourceCodeKind.Regular);
+            TestRoundTripping(node, text, disallowErrors: false);
+            VerifyDirectivesSpecial(node, new DirectiveInfo
+            {
+                Kind = SyntaxKind.ErrorDirectiveTrivia,
+                Status = NodeStatus.IsActive,
+                Text = "version"
+            });
+
+            node.GetDiagnostics().Verify(
+                // (1,8): error CS1029: #error: 'version'
+                // #error version
+                Diagnostic(ErrorCode.ERR_ErrorDirective, "version").WithArguments("version").WithLocation(1, 8),
+                // (1,8): error CS8304: Compiler version: '42.42.42.42424 (<developer build>)'. Language version: 4.
+                // #error version
+                Diagnostic(ErrorCode.ERR_CompilerAndLanguageVersion, "version").WithArguments(GetExpectedVersion(), "4").WithLocation(1, 8)
```

Comment:
I thought you were talking about the formatting ('4' vs. 4) in your previous comment :-S

The reason is because `Parse` uses language version 4:
```C#
   private CSharpParseOptions GetOptions(SourceCodeKind kind, string[] defines)
        {
            return new CSharpParseOptions(languageVersion: LanguageVersion.CSharp4, kind: kind, preprocessorSymbols: defines);
        }
```